x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameTransformCheckpoint.java

```diff
@@ -291,8 +291,14 @@ public static long getBehind(DataFrameTransformCheckpoint oldCheckpoint, DataFra
             }
         }
 
-        for (long[] v : newCheckpoint.indicesCheckpoints.values()) {
-            newCheckPointSum += Arrays.stream(v).sum();
+        for (Entry<String, long[]> entry : newCheckpoint.indicesCheckpoints.entrySet()) {
+            newCheckPointSum += Arrays.stream(entry.getValue()).sum();
```

Comment:
Note that for every brand new shard that has never had a document indexed into it this sum with include a value of -1.  This is not new - the equivalent line in the old code did this too.  Before this PR you could make `operations_behind` go negative by creating a new index with more shards than documents!